File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(n)
    @numbers = (1..n).to_a
  end

  def difference
    square_of_sum - sum_of_squares
  end

  def square_of_sum
    @numbers.reduce(0, &:+) ** 2
  end

  def sum_of_squares
    @numbers.reduce(0) { |acc, i| acc + i ** 2 }
  end
end

module BookKeeping; VERSION = 3; end

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var sum: usize = 0;

    for (1..number + 1) |i| {
        sum += i;
    }

    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    var sum: usize = 0;

    for (1..number + 1) |i| {
        sum += i * i;
    }

    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
